istioctl/cmd/sidecar.go

```diff
@@ -0,0 +1,361 @@
+		Args: func(cmd *cobra.Command, args []string) error {
+			if filename == "" {
+				return fmt.Errorf("expecting a WorkloadGroup artifact file")
+			}
+			if outputName == "" {
+				return fmt.Errorf("expecting an output filename")
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			wg := &clientv1alpha3.WorkloadGroup{}
+			if err := readWorkloadGroup(filename, wg); err != nil {
+				return err
+			}
+			kubeClient, err := kube.NewExtendedClient(kube.BuildClientCmd(kubeconfig, configContext), "")
+			if err != nil {
+				return err
+			}
+
+			if err = createConfig(kubeClient, wg, clusterID, revision, outputName); err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+	generateConfigCmd.PersistentFlags().StringVarP(&filename, "file", "f", "", "filename of the WorkloadGroup artifact")
+	generateConfigCmd.PersistentFlags().StringVarP(&outputName, "output", "o", "", "Name of the tarball to be created")
+
+	generateConfigCmd.PersistentFlags().StringVar(&clusterID, "cluster id", "", "The ID used to identify the cluster")
+	generateConfigCmd.PersistentFlags().StringVar(&revision, "revision", "", "control plane revision (experimental)")
+	return generateConfigCmd
+}
+
+func readWorkloadGroup(filename string, wg *clientv1alpha3.WorkloadGroup) error {
+	f, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return err
+	}
+	if err = yaml.Unmarshal(f, wg); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Creates all the relevant config for the given workload group and cluster
+func createConfig(kubeClient kube.ExtendedClient, wg *clientv1alpha3.WorkloadGroup, clusterID, revision, outputName string) error {
+	temp, err := ioutil.TempDir("./", outputName)
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(temp)
+
+	if err = createClusterEnv(wg, temp); err != nil {
+		return err
+	}
+	if err = createCertificates(kubeClient, temp); err != nil {
+		return err
+	}
+	if err = createMeshConfig(kubeClient, wg, clusterID, revision, temp); err != nil {
+		return err
+	}
+	if err = Tar(temp, outputName); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Write cluster.env into the given directory
+func createClusterEnv(wg *clientv1alpha3.WorkloadGroup, dir string) error {
+	we := wg.Spec.Template.Spec
+	for _, v := range we.Ports {
+		ports = append(ports, fmt.Sprint(v))
+	}
+	// default attributes and service name, namespace, ports, service account, service CIDR
+	clusterEnv := map[string]string{
+		"ISTIO_CP_AUTH":       "MUTUAL_TLS",
+		"ISTIO_INBOUND_PORTS": strings.Join(ports, ","),
+		"ISTIO_NAMESPACE":     wg.Namespace,
+		"ISTIO_PILOT_PORT":    "15012",
+		"ISTIO_SERVICE":       fmt.Sprintf("%s.%s", wg.Name, wg.Namespace),
+		"ISTIO_SERVICE_CIDR":  "*",
+		"SERVICE_ACCOUNT":     we.ServiceAccount,
+	}
+	return ioutil.WriteFile(dir+"/cluster.env", []byte(mapToString(clusterEnv)), tempPerms)
+}
+
+// Get and store the needed certificates
+// TODO: user internal generate-cert
```

Comment:
do you mean use? user internal generate-cert doesn't make alot of sense to me...

Reply:
Typo sorry, was a reminder for myself to switch to the generate-cert package once I figured out what was needed.